src/key.js

```diff
@@ -482,19 +485,29 @@ Key.prototype.getExpirationTime = async function() {
  * - if multiple primary users exist, returns the one with the latest self signature
  * - otherwise, returns the user with the latest self signature
  * @param  {Date} date use the given date for verification instead of the current time
+ * @param  {Object} userId (optional) user ID to get instead of the primary user, if it exists
  * @returns {Promise<{user: module:key.User,
  *                    selfCertification: module:packet.Signature}>} The primary user and the self signature
  * @async
  */
-Key.prototype.getPrimaryUser = async function(date=new Date()) {
-  // sort by primary user flag and signature creation time
+Key.prototype.getPrimaryUser = async function(date=new Date(), userId={}) {
+  if (!this.users.length) {
+    return null;
+  }
+  // sort by userId, primary user flag and signature creation time
   const primaryUser = this.users.map(function(user, index) {
     const selfCertification = getLatestSignature(user.selfCertifications, date);
     return { index, user, selfCertification };
   }).sort(function(a, b) {
     const A = a.selfCertification;
     const B = b.selfCertification;
-    return (A.isPrimaryUserID - B.isPrimaryUserID) || (A.created - B.created);
+    return (
+      (a.user.userId.email === userId.email) - (b.user.userId.email === userId.email) ||
+      (a.user.userId.name === userId.name) - (b.user.userId.name === userId.name) ||
+      (a.user.userId.comment === userId.comment) - (b.user.userId.comment === userId.comment) ||
```

Comment:
I was more thinking of the case where the name and email are not equal but the comment is, and we end up choosing that user based on the comment alone

Reply:
Right, so you're saying that if we have [A (phone), B (desktop)], and we're looking for C (desktop), we should pick the primary user instead of B? That makes sense, unless I guess the user explicitly passes `{ comment: 'desktop' }` as the userid... I don't have a particularly strong opinion on this except that handling all these cases would make the if statement pretty unwieldy ;p

Also, passing a userid which then doesn't exist in the key indicates to me that something has gone wrong, since the same userid should've been used to query the key from a keyring/HKP in the first place, right? It could even be argued that we should throw an error in that case, also as a safeguard to prevent encrypting with the wrong public key, but that seems a bit harsh.

Reply:
Actually now that I think about it I like the idea of being harsh about name and email - if a userId is passed in and we don’t find a user that corresponds to the name or email, I think it’s reasonable to throw an error